server/controllers/authController.js

```diff
@@ -49,4 +50,37 @@ const verifyEmail = async (req, res) => {
     });
 };
 
-export default { signUp, verifyEmail };
+const createUser = async (req, res) => {
+    const { firstName, lastName, email } = req.body;
+    const password = helpers.random8Password();
+    const rawPassword = password;
+
+    const auth_first_name = req.user.firstName;
+    const auth_last_name = req.user.lastName;
+    const name = `${auth_first_name} ${auth_last_name}`;
+
+    await User.query().insert({
+        firstName,
+        lastName,
+        password,
+        email,
+        by_admin: true
+    });
+
+    await new Mail('invite-mail')
```

Comment:
Let's have a little refactor. Just like the welcome email, let's move this to the user model. have a look:

```js

class User extends Model {
   ....
    async sendWelcomeMail () {
        await new Mail('welcome-mail')
            .to(this.email, this.firstName)
            .data({
                name: this.firstName,
                url: `${config.server.url}/api/v1/auth/verify/${
                    this.email_confirm_code
                }`
            })
            .subject('Welcome Onboard')
            .send();
    }

    async sendInviteMail({ data }) {
        await new Mail('invite-mail')
            .to(this.email, this.firstName)
            .data({
                name: this.firstName,
                email: this.email,
                invitee_name: data.name,
                password: this.generateRawPassword()
            })
            .subject('Hello-Books invite mail')
            .send();
    }

    async $afterInsert(context) {
        await super.$afterInsert(context);

        if (! this.by_admin) {
            this.sendWelcomeMail()
        }
    }

}
```

Now, with this refactor, the methods `sendWelcomeMail` and `sendInviteMail` become reusable in future, if, for example if we needed to resend email verification or resend welcome email from admin dashboard, we can easily reuse this function in a controller.